app/src/components/features/rules/RulePairs/Pairs/Rows/RowsMarkup/RequestSourceRow/index.js

```diff
@@ -142,7 +119,7 @@ const RequestSourceRow = ({
       wrap={false}
     >
       <Col className="shrink-0">
-        <RQDropdown overlay={sourceKeyOptions} disabled={isInputDisabled}>
+        <RQDropdown overlay={renderSourceKeys()} disabled={isInputDisabled}>
```

Comment:
Calling the function in each render cycle is not optimized.
Rather, you should use `useMemo`.
```js
const sourceKeysMenu = useMemo(() => (
  <Menu>
  </Menu>
), [sourceKeys]);
```